smact/properties.py

```diff
@@ -157,3 +160,56 @@ def compound_electroneg(
         print("Geometric mean = Compound 'electronegativity'=", compelectroneg)
 
     return compelectroneg
+
+
+def valence_electron_count(compound: str) -> float:
+    """
+    Calculate the Valence Electron Count (VEC) for a given compound.
+
+    This function parses the input compound, extracts the elements and their
+    stoichiometries, and calculates the VEC using the valence electron data
+    from SMACT's Element class.
+
+    Args:
+        compound (str): Chemical formula of the compound (e.g., "Fe2O3").
+
+    Returns:
+        float: Valence Electron Count (VEC) for the compound.
+
+    Raises:
+        ValueError: If an element in the compound is not found in the valence data.
+    """
+    from typing import Dict
+
+    def parse_formula(formula: str) -> Dict[str, int]:
+        pattern = re.compile(r"([A-Z][a-z]*)(\d*)")
+        elements = pattern.findall(formula)
+        element_stoich: Dict[str, int] = defaultdict(int)
+        for element, count in elements:
+            count = int(count) if count else 1
+            element_stoich[element] += count
+        return element_stoich
+
+    def get_element_valence(element: str) -> int:
+        try:
+            return smact.Element(element).num_valence_modified
+        except AttributeError:

```

Comment:
Can I ask why this is an AttributeError?
If the Element doesn't exist, the first error that gets raised would be `NameError: Elemental data for X not found.`
Then another potential error that can be raised is a `TypeError: unsupported operand type(s) for *: 'int' and 'NoneType'` which occurs when the Element does exist but has no valence data and returns `None` as its valence e.g. Lr